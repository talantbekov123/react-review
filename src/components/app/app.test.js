import React from 'react'
import renderer from 'react-test-renderer'
import App from './app'
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import {createLogger} from "redux-logger";
import {applyMiddleware, createStore} from "redux";
import reducer from "../../reducers";

const middleware = [thunk]
if (process.env.NODE_ENV !== 'production') {
  middleware.push(createLogger())
}

const store = createStore(reducer, applyMiddleware(...middleware))


it('renders without crashing', () => {
  const tree = renderer
      .create(
          <Provider store={store}>
            <Router>
              <App />
            </Router>
          </Provider>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
})
