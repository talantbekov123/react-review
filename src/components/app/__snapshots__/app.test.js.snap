// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="top-menu"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button>
          <span>
            Главная
          </span>
        </button>
      </a>
      <a
        href="/profile"
        onClick={[Function]}
      >
        <button>
          <span>
            Профиль
          </span>
        </button>
      </a>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button>
          <span>
            Логин
          </span>
        </button>
      </a>
    </div>
  </header>
  <div
    className="content"
  >
    <h2>
      Главная страница
    </h2>
  </div>
</div>
`;
